### PR TITLE
Issue 61

### PR DESCRIPTION
**Pull request by carlosTorres07**

### CODE REVIEW

File: dal/src/main/java/com/craftsmanship/tfm/dal/grpc/server/GrpcServer.java [hunk 1]
```diff
@@ -156,10 +156,10 @@ public void delete(DeleteItemRequest request, io.grpc.stub.StreamObserver<Delete
 	private GrpcItem getGrpcItemFromItem(Item item) {
 		return GrpcItem.newBuilder()
         		                  .setId(item.getId())
-        		                  .setName(item.getDescription()).build();
+        		                  .setName(item.getName()).build();
```
Comment (kikogolk) [inline dal/src/main/java/com/craftsmanship/tfm/dal/grpc/server/GrpcServer.java]: You are missing price and quantity?

File: dal/src/main/java/com/craftsmanship/tfm/dal/grpc/server/GrpcServer.java [hunk 2]
```diff
@@ -86,7 +86,7 @@ public ItemPersistenceServiceImpl(DataAccess dataAccess) {
     public void create(CreateItemRequest request, io.grpc.stub.StreamObserver<CreateItemResponse> responseObserver) {
       logger.info("CREATE RPC CALLED");
       GrpcItem grpcItem = request.getItem();
-      Item item = new Item(grpcItem.getName());
+      Item item = new Item(grpcItem.getName(), grpcItem.getPrice(), grpcItem.getQuantity());
```
Comment (kikogolk) [inline dal/src/main/java/com/craftsmanship/tfm/dal/grpc/server/GrpcServer.java]: why not use getItemFromGrpcItem ?

File: dal/src/test/java/com/craftsmanship/tfm/dal/model/ItemEntityTest.java [hunk 3]
```diff
@@ -45,13 +45,13 @@ public void tearDown() throws Exception {
 	@Test
 	public void whenSaveItem_thenReturnItem() {
 		Item savedItem = this.entityManager.persistAndFlush(item);
-		assertThat(savedItem.getDescription()).isEqualTo("Item description");
+		assertThat(savedItem.getName()).isEqualTo("Item description");
```
Comment (kikogolk) [inline dal/src/test/java/com/craftsmanship/tfm/dal/model/ItemEntityTest.java]: If you are comparing names... why not the rest of attributes?

**Review by kikogolk — CHANGES_REQUESTED** [review]

File: dal/src/main/java/com/craftsmanship/tfm/dal/grpc/server/GrpcServer.java (hunk 2 above)
Comment (carlosTorres07) [reply dal/src/main/java/com/craftsmanship/tfm/dal/grpc/server/GrpcServer.java]: I agree with the comments.
I create a new commit with modifications.

File: dal/src/main/java/com/craftsmanship/tfm/dal/grpc/server/GrpcServer.java (hunk 1 above)
Comment (carlosTorres07) [reply dal/src/main/java/com/craftsmanship/tfm/dal/grpc/server/GrpcServer.java]: Done

File: dal/src/test/java/com/craftsmanship/tfm/dal/model/ItemEntityTest.java (hunk 3 above)
Comment (carlosTorres07) [reply dal/src/test/java/com/craftsmanship/tfm/dal/model/ItemEntityTest.java]: Done

File: dal/src/main/java/com/craftsmanship/tfm/dal/grpc/server/GrpcServer.java [hunk 4]
```diff
@@ -29,144 +29,144 @@
...
+            responseObserver.onNext(response);
+            responseObserver.onCompleted();
+        }
+
+        @Override
+        public void update(UpdateItemRequest request, io.grpc.stub.StreamObserver<UpdateItemResponse> responseObserver) {
+            logger.info("UPDATE RPC CALLED");
+
+            // TODO: right now, if the id does not exists, the item is creted, Should we
+            // raise an error?
```
Comment (alb3rtobr) [inline dal/src/main/java/com/craftsmanship/tfm/dal/grpc/server/GrpcServer.java:132]: I think we should raise an exception, but I think we can live without it by the moment. There are too many exceptions we have to add...

**Review by alb3rtobr — APPROVED** [review]

**Review by kikogolk — APPROVED** [review]

**Pull request merged**